docs/faq.rst

```diff
@@ -27,7 +27,8 @@ Array                               Any supported type
 Enumeration                         Any NumPy integer type                          Read/write as integers
 References                          Region and object
 Variable length array               Any supported type                              See :ref:`Special Types <vlen>`
-=========================           ============================================    ======================
+Any other NumPy dtype               Map to opaque type with specific tag            You must register types manually
+=========================           ============================================    ================================
```

Comment:
I wouldn't add this here, I'd add a section in the unsupported types noting that users can use the opaque type to store other data and can tag it (and put a comment in the opaque type line referring to that section).